### DISCUSSION TITLE
Option Bytes for bootloader

### DISCUSSION THREAD

**Discussion by MadZoids** [stm32duino/Arduino_Core_STM32#1657] [Q&A] [answered]
Does option bytes always reset to factory configuration on new compile on generic board variants?
Can we change the option bytes from sketch level?

**Comment by ABOSTM** [comment, marked as answer]
Hi @MadZoids,

> Does option bytes always reset to factory configuration on new compile on generic board variants?

No, Option bytes remains after reset or power OFF/ON. And it is not changed on compilation nor during download of binary in flash.

> Can we change the option bytes from sketch level?

There is no dedicated API to do so in Arduino.
Nevertheless you can still make use of STM32 HAL/LL or even direct Register access (all are accessible from Arduino sketch level)
Like: 
HAL_FLASH_OB_Unlock()
HAL_FLASHEx_OBProgram()
See for example https://github.com/STMicroelectronics/STM32CubeL4/tree/master/Projects/NUCLEO-L412KB/Examples/FLASH/FLASH_WriteProtection
